django/core/files/storage.py

```diff
@@ -217,7 +217,7 @@ def directory_permissions_mode(self):
     def _open(self, name, mode='rb'):
         return File(open(self.path(name), mode))
 
-    def _save(self, name, content):
+    def _save(self, name, content, allow_overwrite=False):
```

Comment:
The parameter to this method seems odd to me. As it stands, no caller is explicitly using the parameter in a call to the method and I can't see many legitimate uses for it in derived classes. I think you're effectively just using the default value as a space to store a class-scope variable. Would it be simpler to just assign a member on the class? Or on the instance, if you prefer.

Reply:
>I think you're effectively just using the default value as a space to store a class-scope variable.

Yeah, I think that this is a fair assessment.

As a class-level attribute I think that [it would look fairly reasonable](https://github.com/jonprindiville/django/compare/ticket_28144...ticket_28144--class-attribute) but I wonder about naming it `ALLOW_OVERWRITE`. Really in order to do the overwriting you need to modify `get_available_name` as well (since it is called before `_save`.)

[Another sketch I did was to pull out the `which flags should we pass to os.open` into a method](https://github.com/jonprindiville/django/compare/ticket_28144...jonprindiville:ticket_28144--method-hook) which could be overridden by a subclasser. My example `OverwritingStorage` just omits the O_EXCL from its list but it could also call super and then mask out the O_EXCL.

You have thoughts on either of those approaches?

(You mention setting it on the instance as a possibility, as well: did you mean accepting an `allow_overwrite` kwarg in `FileSystemStorage.__init__` or just that init hardcode a `self._allow_overwrite = False`?)

Reply:
Having an overridable method seems like the most orthodox OOP solution (it's what a Java programmer would do :-) ) but I'm not convinced it really gives a useful abstraction: by coincidence it's the right place to make this one change, but I'm not sure there's a useful class of future modifications it opens up, so it feels like overkill to me.

My thought with an instance variable was just to set it in the constructor in the base class, and overwrite it in the subclass constructor (not exposing it as a kwarg). I'm not sure there's any advantage to this; I think I was thinking about this because it's what I'd do in C++.

I don't have a particularly strong feeling on this. I think if I were writing it I'd go with the class-level attribute.